src/web/security/core/src/main/java/org/geoserver/security/web/url/URLChecksPage.java

```diff
@@ -112,14 +117,43 @@ public void onSubmit(AjaxRequestTarget target, Form form) {
                         addFeedbackPanels(target);
                     }
 
-                    private AbstractURLCheck getMatchingRule() throws IOException {
+                    private URLChecker getMatchingRule() throws IOException {
                         String test = testInput.getInput();
                         List<AbstractURLCheck> checks = getUrlCheckDAO().getChecks();
+                        // Check 1: Check using test URL provided
                         for (AbstractURLCheck check : checks) {
                             if (check.isEnabled()) {
                                 if (check.confirm(test)) return check;
                             }
                         }
+                        // Check 2: Check using normalized test URL if different
+                        try {
+                            // URL normalization to resolve relative paths prior to checking
+                            String normalized = new URI(test).normalize().toString();
+                            if (!normalized.equals(test)) {
+                                for (AbstractURLCheck check : getUrlCheckDAO().getChecks()) {
```

Comment:
Why go to the DAO again, you already have the list of checks?

Reply:
I was trying to get all the checks, the geotools ones also.

Previously we only had the geoserver checks.